maskrcnn_benchmark/data/build.py

```diff
@@ -36,16 +37,16 @@ def build_dataset(dataset_list, transforms, dataset_catalog, is_train=True):
         # for COCODataset, we want to remove images without annotations
         # during training
         if data["factory"] == "COCODataset":
-            args["remove_images_without_annotations"] = is_train
+            args["remove_images_without_annotations"] = mode != DatasetMode.TEST
         if data["factory"] == "PascalVOCDataset":
-            args["use_difficult"] = not is_train
+            args["use_difficult"] = mode == DatasetMode.TEST
         args["transforms"] = transforms
         # make dataset from factory
         dataset = factory(**args)
         datasets.append(dataset)
 
     # for testing, return a list of datasets
-    if not is_train:
+    if mode != DatasetMode.TEST:
```

Comment:
Even though not really the best thing to do, I believe in most cases we simply evaluate on the test dataset after N iterations, so I think that we can remove the `VAL` part altogether.

Reply:
I added another boolean flag instead for controlling the way of data loader creating:
https://github.com/facebookresearch/maskrcnn-benchmark/pull/828/files#diff-48c338613bdbf422235cdb2ef17201f7R77